composables/useSearch.ts

```diff
@@ -0,0 +1,64 @@
+// TODO: type SearchResult and SearchResponse from `Quang19992/momoka` GraphQL api
+
+interface SearchResult {
+  id: number;
+  name: string;
+  publisher: {
+    id: string;
+    name: string;
+  };
+  type: {
+    id: string;
+    name: string;
+    color: string;
+  };
+  status: string;
+  image_url: string;
+}
+
+interface SearchResponse {
+  hits: SearchResult[];
+}
+
+export default function (query: Ref<string>, limit = 5) {
+  const data = ref<SearchResponse | null>(null);
+  const error = ref();
+  const pending = ref<Boolean>();
+
+  async function doFetch() {
+    // shows no result when query is <empty string>
+    if (query.value === "") {
+      data.value = null;
+      return;
+    }
+
+    // initialise values
+    pending.value = true;
+    data.value = {
+      hits: [],
+    };
+
+    const res = await useFetch<SearchResponse>(
+      computed(
+        () =>
+          `${useRuntimeConfig().public.search_endpoint}?q=${
+            query.value
+          }&limit=${limit}`
+      ),
+      {
+        headers: {
+          Authorization: `Bearer ${useRuntimeConfig().public.search_token}`,
+        },
+      }
+    );
```

Comment:
should made a `search({query: String}): Title[]` function for easier change in the future

Reply:
this was implemented by the default `useSearch` composables (based on [how Nuxt handle composable file name as function name](https://nuxt.com/docs/guide/directory-structure/composables)), which could be called with 
```typescript
useSearch(query: Ref<string>, limit?)
```
which then return an object consist `{ data, pending, error }`